Improve diagnostics for WPT failures

SimonSapin opened the pull request:
* Include the full output (including stdout/stderr) in the intermittent-filtered log
* Print the intermittent-filtered log at the end of the main log (which is one less click to reach from Taskcluster’s task view, compared to other task artifacts)
* <del>Fail with a specific message when a reftest screenshot is entirely white</del> (This caused over a hundred unexpected results. A few of them in reftests that use `about:blank` as a reference.)
* For failing reftests, add a message if the whole screenshot is a solid color, to help recognize instances of https://github.com/servo/servo/issues/24726

  ```
    ▶ FAIL [expected PASS] /css/CSS2/box-display/root-box-003.xht
    │   → /css/CSS2/box-display/root-box-003.xht 54a9df64f1476dd12020019d7cf22ac34d727bc0
    │   → /css/CSS2/box-display/root-box-003-ref.xht 636eb693bc214b6e1c64e6566c48e69e6777b946
    └   → Screenshot is solid color 0xFFFFFF for /css/CSS2/box-display/root-box-003.xht
  ```
  (The last line is new.)

SimonSapin committed:
Include test output in filtered WPT logs

SimonSapin commented:
@bors-servo try=wpt

bors-servo commented:
:hourglass: Trying commit 308e4f707e74eeefac848548ee796d8818220260 with merge c831b6fa59113eb7a0e1ed6ba13adf8fdd224c7b...

servo-wpt-sync commented:
Opened new PR for upstreamable changes.

Completed upstream sync of web-platform-test changes at https://github.com/web-platform-tests/wpt/pull/20549.

bors-servo commented:
:broken_heart: Test failed - [status-taskcluster](https://community-tc.services.mozilla.com/tasks/groups/OdN1V6J7TjWVVfeM0D-NoQ)

SimonSapin commented:
@bors-servo try=wpt

bors-servo commented:
:hourglass: Trying commit 1e1f191206a9e3c31f36ee3ded7a1768fd7ca8ba with merge dd7003b4888072e1bce0114a5f03acf65585e20f...

servo-wpt-sync commented:
Transplanted upstreamable changes to existing PR.

Completed upstream sync of web-platform-test changes at w3c/web-platform-tests#20549.

bors-servo commented:
:broken_heart: Test failed - [status-taskcluster](https://community-tc.services.mozilla.com/tasks/groups/bL0-gdEBRFiDB_xs5KZ9vw)

SimonSapin committed:
wptrunner: detect solid color screenshots for failing reftests

The output looks like this:

```
  ▶ FAIL [expected PASS] /css/CSS2/box-display/root-box-003.xht
  │   → /css/CSS2/box-display/root-box-003.xht 54a9df64f1476dd12020019d7cf22ac34d727bc0
  │   → /css/CSS2/box-display/root-box-003-ref.xht 636eb693bc214b6e1c64e6566c48e69e6777b946
  └   → Screenshot is solid color 0xFFFFFF for /css/CSS2/box-display/root-box-003.xht
```

SimonSapin committed:
Improve formatting of multi-line "message" in Servo’s WPT log formatter

servo-wpt-sync commented:
Transplanted upstreamable changes to existing PR.

Completed upstream sync of web-platform-test changes at w3c/web-platform-tests#20549.

SimonSapin commented:
@bors-servo try=wpt

bors-servo commented:
:hourglass: Trying commit 10a35725b6651083237fdedb19c86d2bbd5bda03 with merge e0124781b0cd3d8a7cde319aace7fb57b713cbf7...

bors-servo commented:
:broken_heart: Test failed - [status-taskcluster](https://community-tc.services.mozilla.com/tasks/groups/Bp3W16eTRz60SR6HK6IV2Q)

SimonSapin committed:
Add a test counter in WPT’s non-interactive output

servo-wpt-sync commented:
Transplanted upstreamable changes to existing PR.

Completed upstream sync of web-platform-test changes at w3c/web-platform-tests#20549.

SimonSapin commented:
Perfect, the two filtered unexpected results demonstrate the changes in this PR:

* https://community-tc.services.mozilla.com/tasks/MFCiPSOsQB2FWyQGBNU4cA/runs/0/logs/https%3A%2F%2Fcommunity-tc.services.mozilla.com%2Fapi%2Fqueue%2Fv1%2Ftask%2FMFCiPSOsQB2FWyQGBNU4cA%2Fruns%2F0%2Fartifacts%2Fpublic%2Flogs%2Flive.log#L7603-7610
* https://community-tc.services.mozilla.com/tasks/MFCiPSOsQB2FWyQGBNU4cA/runs/0/logs/https%3A%2F%2Fcommunity-tc.services.mozilla.com%2Fapi%2Fqueue%2Fv1%2Ftask%2FMFCiPSOsQB2FWyQGBNU4cA%2Fruns%2F0%2Fartifacts%2Fpublic%2Flogs%2Flive.log#L7603-7610

This is ready for review.

r? @jdm
CC @jgraham for changes in ` tests/wpt/web-platform-tests/tools/wptrunner`

servo-wpt-sync commented:
Transplanted upstreamable changes to existing PR.

Completed upstream sync of web-platform-test changes at w3c/web-platform-tests#20549.

jdm commented:
@bors-servo r+
Nice improvement!

bors-servo commented:
:pushpin: Commit d0785e0 has been approved by `jdm`

<!-- @bors-servo r=jdm d0785e0ba5b22d9916a49f52e71e4198df9b5cb6 -->

bors-servo commented:
:hourglass: Testing commit d0785e0ba5b22d9916a49f52e71e4198df9b5cb6 with merge ef2512d10488b96c04e410a3ee7c2543fa3393ed...

bors-servo commented:
:broken_heart: Test failed - [status-taskcluster](https://community-tc.services.mozilla.com/tasks/groups/E7WrIZyvRCS7-EbGI1Joeg)

jdm commented:
@bors-servo retry
* #22212

bors-servo commented:
:hourglass: Testing commit d0785e0ba5b22d9916a49f52e71e4198df9b5cb6 with merge 64b7a31cd71e3862b9ff9a4bec1eb6f47a747f55...

bors-servo commented:
:broken_heart: Test failed - [status-taskcluster](https://community-tc.services.mozilla.com/tasks/groups/ao3TkAVkQjyD4JaCbe_XxA)

servo-wpt-sync commented:
Transplanted upstreamable changes to existing PR.

Completed upstream sync of web-platform-test changes at w3c/web-platform-tests#20549.

jdm commented:
@bors-servo try=wpt

bors-servo commented:
:hourglass: Trying commit 46b122f3b31788da4dcd9a102a76fdd0b18bffd0 with merge 955431f8a2c6a8decbf364de27ba1b7b4972e1fb...

bors-servo commented:
:broken_heart: Test failed - [status-taskcluster](https://community-tc.services.mozilla.com/tasks/groups/Yu3F8qfnQQeJjEzSgndB8A)

SimonSapin commented:
I think what happened in https://github.com/servo/servo/pull/24981#issuecomment-560954876 was that I forced-push from the wrong machine an earlier version of this branch. I’ve canceled this last `try=wpt` run since it only tests Linux which I’d already tested on the more recent version of the branch, and there seems to also be some Python errors on macOS. I’m testing on the `try-taskcluster` branch at the moment.

servo-wpt-sync commented:
Transplanted upstreamable changes to existing PR.

Completed upstream sync of web-platform-test changes at w3c/web-platform-tests#20549.

SimonSapin committed:
Fix Python Unicode error on macOS CI

servo-wpt-sync commented:
Transplanted upstreamable changes to existing PR.

Completed upstream sync of web-platform-test changes at w3c/web-platform-tests#20549.

SimonSapin commented:
Alright, I think this should be good now. r? @jdm, since you might have reviewed the obsolete branch

jdm commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit 14f049d has been approved by `jdm`

<!-- @bors-servo r=jdm 14f049ddd1ff0801f12f7a2781a2caaaa5b9e81f -->

bors-servo commented:
:hourglass: Testing commit 14f049ddd1ff0801f12f7a2781a2caaaa5b9e81f with merge d99850ffa8626674ae492c287c04950abc52b088...

bors-servo commented:
:sunny: Test successful - [status-taskcluster](https://community-tc.services.mozilla.com/tasks/groups/UxrM6ax7R2y4o20CBu57Lw)
Approved by: jdm
Pushing d99850ffa8626674ae492c287c04950abc52b088 to master...
